src/mercury_engine_data_structures/formats/bmdefs.py

```diff
@@ -0,0 +1,100 @@
+import construct
+from construct import (
+    Const,
+    Construct,
+    Enum,
+    Float32l,
+    Int8ul,
+    Int16ul,
+    Int32ul,
+    Struct,
+)
+
+from mercury_engine_data_structures.common_types import StrId, make_vector
+from mercury_engine_data_structures.formats import BaseResource
+from mercury_engine_data_structures.game_check import Game
+
+Status = Enum(
+    Int8ul,
+    DISABLED=0,
+    ENABLED=1,
+)
+
+EnemiesStruct = Struct(
+    "enemy_name" / StrId,
+    "areas" / make_vector(Struct(
+        "area_name" / StrId,
+        "layers" / make_vector(Struct(
+            "layer_name" / StrId,
+            "states" / make_vector(Struct(
+                "type" / StrId,
+                value=construct.Switch(
+                    construct.this.type,
+                    {
+                        'COMBAT': Struct(
+                            "unk1" / Int32ul,
+                            "priority" / Int32ul,
+                            "file_name" / StrId,
+                            "fade_in" / Float32l,
+                            "start_delay" / Float32l,
+                            "volume" / Float32l,
+                            "unk2" / Int32ul,
+                            "unk3" / Int32ul,
+                            "unk4" / Int32ul,
+                            "status" / Status,
+                            "float1" / Int32ul,
+                            "inner_states" / make_vector(Struct(
+                                "type" / StrId,
+                                "unk1" / Float32l,
+                            ))
+
+                        ),
+                        'DEATH': Struct(
+                            "unk1" / Int32ul,
+                            "priority" / Int32ul,
+                            "file_name" / StrId,
+                            "start_delay" / Float32l,
+                            "fade_out" / Float32l,
+                            "volume" / Float32l,
+                            "unk2" / Int32ul,
+                            "unk3" / Int32ul,
+                            "unk4" / Int32ul,
+                            "status" / Status,
+                            "float1" / Float32l,
+                            "unk5" / Int32ul,
+                        ),
```

Comment:
When I initially saw the red marked length field, I had some trouble to figure out where the second thing starts. As you can see there is two times the `DEATH` string very close together. Eventually, when I had it correct I was too tired to clean it up (therefore there were just `Byte` blocks before)
![image](https://github.com/user-attachments/assets/f9e4fd48-66c9-4c76-b4f2-9c111df134f9)

The way you have it now means that it doesn't need the whole `Switch` at all. `unk5` here is just the length field of what is called `inner_states` which makes both `Struct`s identical.

...**unless** you really think that after `file_name` one really has `fade_in` and then `start_delay` while the other has `start_delay` and then `fade_out` but that would be really weird.

Reply:
I swear that's what i tested. Maybe I'm wrong though.

Reply:
It's also possible I was misinterpreting it. So it'd be any easy fix to just name the field `fade`

Reply:
Ok so I retested this again and it is actually different between the two. Gonna try to simplify if possible here.

Reply:
Ok, wow. Both follow the exact same format but the two fields after the file_name have a different meaning?

Reply:
Yeah that's the only difference. I'm not sure if there is a good way to simplify this. Could make structs for everything above and below those, but it seems extra